geniza/corpus/views.py

```diff
@@ -318,6 +318,23 @@ def get_context_data(self, **kwargs):
         """extend context data to add page metadata"""
         context_data = super().get_context_data(**kwargs)
         images = self.object.iiif_images(with_placeholders=True)
+
+        # logic for which panels to show or disable by default: images always shown if available,
+        # then priority is translations over transcriptions. show both of the latter two if no imgs
+        default_shown = [
+            "images" if self.object.has_image() else None,
+            "translation" if self.object.has_translation() else None,
+            "transcription"
+            if self.object.has_transcription()
+            and (not self.object.has_translation() or not self.object.has_image())
+            else None,
+        ]
+        disabled = [
+            "images" if not self.object.has_image() else None,
+            "translation" if not self.object.has_translation() else None,
+            "transcription" if not self.object.has_transcription() else None,
+        ]
```

Comment:
❤️ love seeing this configured in the view, thank you!

Can we avoid calling these `has_` methods twice? My quick thoughts on options: either convert them to cached properties, or checking for which ones are disabled first and then using _that_ to determine the defaults.

Reply:
ok, actually - rereading, the code and thinking about the logic for the defaults (which is a little hard to follow), I think refactoring and basing it on what is / is not disabled should simplify. Instead of that complicated  if, and, or statement, you just need a list of which panels are available (based on which ones are not), list them in the order of priority, and then take the first two. I think that also removes the need for the list comprehension filter you're doing on line 362 to filter out the `None` values.